### PR TITLE
Did functionality for delete interview and delete notes in the notes screen

### PR DESCRIPTION
**Pull request by uhalder**
- Added a relationship between interview and notes
- Added use cases appropriately
- Connected the use cases in the view model, updated the UI
- Created a dropdown menu for delete interview and delete notes

### CODE REVIEW

File: app/src/main/kotlin/com/twain/interprep/data/dao/NoteDAO.kt [hunk 1]
```diff
@@ -37,6 +37,9 @@ interface NoteDAO {
     @Query("SELECT * FROM interview")
     fun getAllInterviewNoteMap(): Flow<List<InterviewWithNotes>>
 
+    @Query("DELETE FROM note where interviewId = :interviewId")
+    fun deleteNotesForInterview(interviewId: Int)
```
Comment (Audienix) [inline app/src/main/kotlin/com/twain/interprep/data/dao/NoteDAO.kt]:
```suggestion
    fun deleteInterviewNotes(interviewId: Int)
```

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
     <uses-permission android:name="android.permission.POST_NOTIFICATIONS" />
     <application
-        android:allowBackup="true"
+        android:allowBackup="false"
```
Comment (Audienix) [inline app/src/main/AndroidManifest.xml]: Any particular reason for this change?

File: app/src/main/kotlin/com/twain/interprep/data/repository/NoteRepositoryImpl.kt [hunk 3]
```diff
@@ -38,4 +38,8 @@ class NoteRepositoryImpl(private val noteDAO: NoteDAO) : NoteRepository {
         noteDAO.deleteNote(note)
     }
 
+    override suspend fun deleteNotesForInterview(interview: Interview) {
+        noteDAO.deleteNotesForInterview(interview.interviewId)
+    }
+
 }
```
Comment (Audienix) [inline app/src/main/kotlin/com/twain/interprep/data/repository/NoteRepositoryImpl.kt]: insert new line at end-of-file ?

File: app/src/main/kotlin/com/twain/interprep/data/dao/NoteDAO.kt [hunk 4]
```diff
@@ -37,6 +37,9 @@ interface NoteDAO {
...
+
     data class InterviewWithNotes(
```
Comment (Audienix) [inline app/src/main/kotlin/com/twain/interprep/data/dao/NoteDAO.kt:43]: Is this data class now not being used after this introduction of Foreign Key and deleting Notes by CASCADE ?
If this is not used anywhere, can we delete this class?

File: app/src/main/kotlin/com/twain/interprep/domain/usecase/note/DeleteNotesForInterviewUseCase.kt [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+import com.twain.interprep.data.model.Interview
+import com.twain.interprep.domain.repository.NoteRepository
+
+class DeleteNotesForInterviewUseCase(private val noteRepository: NoteRepository) {
+    suspend operator fun invoke(
+        interview: Interview
+    ) = noteRepository.deleteNotesForInterview(interview)
+
+}
```
Comment (Audienix) [inline app/src/main/kotlin/com/twain/interprep/domain/usecase/note/DeleteNotesForInterviewUseCase.kt]: insert new line at end-of-file ?

**Review by Audienix — APPROVED** [review]

File: app/src/main/kotlin/com/twain/interprep/data/dao/NoteDAO.kt (hunk 1 above)
Comment (uhalder) [reply app/src/main/kotlin/com/twain/interprep/data/dao/NoteDAO.kt]: This is fixed

File: app/src/main/kotlin/com/twain/interprep/data/repository/NoteRepositoryImpl.kt (hunk 3 above)
Comment (uhalder) [reply app/src/main/kotlin/com/twain/interprep/data/repository/NoteRepositoryImpl.kt]: This is fixed

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (uhalder) [reply app/src/main/AndroidManifest.xml]: I did this for testing. Now I changed it back

File: app/src/main/kotlin/com/twain/interprep/data/dao/NoteDAO.kt (hunk 4 above)
Comment (uhalder) [reply app/src/main/kotlin/com/twain/interprep/data/dao/NoteDAO.kt:43]: This is used by NotesViewModel to read Notes for Interview.

File: app/src/main/kotlin/com/twain/interprep/domain/usecase/note/DeleteNotesForInterviewUseCase.kt (hunk 5 above)
Comment (uhalder) [reply app/src/main/kotlin/com/twain/interprep/domain/usecase/note/DeleteNotesForInterviewUseCase.kt]: this is fixed

**Pull request merged**